### ISSUE TITLE
proposal: in-built micro pull request (ish) system

### ISSUE THREAD

**Issue by texonidas** [codenetwork/CodeNetworkTV#6]
Not sure if this would even be feasible, but add functionality that allows people watching the live stream to submit changes that the streamer can accept and merge via the streaming client. Which would likely also require a secondary window for changes to be written in, then either highlighting or pointing to where that code would go.

**Comment by Lanzafame** [comment]
Don't really have this entirely worked out in my head but I feel there should be a way that utilising a second `git remote` address that we have on our own servers. We could allow for viewers to clone and then push changes back up. Then the streamer could diff the contributions using regular `git`. 

On a side note, this would probably be really distracting for the streamer and would sort of defeat the purpose of them explaining what they are coding to the audience, if some anonymous (to the other viewers) person is contributing/changing the direction of the session.

With some more thought as to how this would work, I think it would be awesome.

**Comment by texonidas** [comment]
Yeah I was thinking of it being more of an option for when you hit a wall.
i.e. "I'm stuck, can anyone see where I fucked up?"

I think it would likely be a separate piece of the client, not directly
visible, almost like a hamburger menu with a list of potential commits to
accept, rather than directly up in your face.

**Comment by Lanzafame** [comment]
Yeah that sounds good. I reckon if we make it that the streamer can toggle it on/off according to whether they want live contributions, that would work well.

**Comment by hjubb** [comment]
If it's something simple enough gists have full git capabilities to my knowledge? Or I could at least check out and modify code in a gist. I like the idea of a toggle it could almost be used as a learning resource as well but whether or not people would prefer to work in their own IDEs etc to make big enough changes to code / would need to still run tests and everything if they have that in place but sounds definitely cool

**Comment by Lanzafame** [comment]
We could use something like this so that the streamer can use whatever IDE they want: https://github.com/tillberg/gut
